### PR TITLE
Allow pyproject.toml as configuration file

### PR DESCRIPTION
**Pull request by kasium**

### CODE REVIEW

**Comment by kasium** [comment]
@pappasam can you please have a look? I didn't include tests and docs by purpose, so that you can have a first look instead

**Comment by pappasam** [comment]
I love this idea! Totally on board with it. That said, there is some TLC we'll need to do to explicitly handle new versions of `tomlkit` and `mypy` and make the tests pass. A lot has happened in the community since the last time this project has been updated. You've motivated me to focus on that effort again, I'll let you know once that work is done.

**Comment by pappasam** [comment]
This is the relevant branch btw, still a work in progress: https://github.com/pappasam/toml-sort/tree/remove-click

**Comment by kasium** [comment]
Cool, let me know when your work is done. I'll then make my code running on the new base, add test and add docs (which I could also do in parallel)

**Comment by pappasam** [comment]
@kasium ok, I've finished migrating from click to argparse. Latest changes are in `main`!

**Comment by kasium** [comment]
@pappasam during local testing I get this error
```
Traceback (most recent call last):
  File "/root/git/toml-sort/.venv/bin/toml-sort", line 5, in <module>
    cli()
TypeError: cli() missing 1 required positional argument: 'arguments'
```

Seems like, that the entry point calls cli without sys.argv. So the argument of cli() should be optional and if non is given, fallback to sys.argv

**Comment by pappasam** [comment]
> @pappasam during local testing I get this error
> 
> ```
> Traceback (most recent call last):
>   File "/root/git/toml-sort/.venv/bin/toml-sort", line 5, in <module>
>     cli()
> TypeError: cli() missing 1 required positional argument: 'arguments'
> ```
> 
> Seems like, that the entry point calls cli without sys.argv. So the argument of cli() should be optional and if non is given, fallback to sys.argv

I've made the argument optional and have removed the unnecessary `entrypoint` function (we now just target `cli`). So should be resolved!

**Comment by kasium** [comment]
@pappasam can you please have a look on the current implementation? If it's fine I'll add docs and tests

**Review by pappasam — COMMENTED** [review]
Looks generally good! Not sure we need to return a namespace value from `_load_config`, any reason you chose to? Figure we can just use a dictionary

**Comment by pappasam** [comment]
After a bit more thinking, now that we're using argparse, you'll want to do the following here:

1. Strictly validate all types from the config file. Argparse will not provide any validation for us.
2. Set <https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.set_defaults> on the parser using the configuration file's values
3. Parse the arguments as before.

File: toml_sort/cli.py [hunk 1]
```diff
@@ -151,6 +167,11 @@ def cli(  # pylint: disable=too-many-branches
         print(get_version())
         sys.exit(0)
 
+    config = _load_config()
+    for key, value in vars(args).items():
```
[the remark this answers is not in the document]
Comment (kasium) [reply toml_sort/cli.py:171]: Replaced by set_defaults

**Comment by kasium** [comment]
@pappasam I added a kind of validation, which checks if the keys have the right type and also if the section includes unknown config keys

**Comment by kasium** [comment]
@pappasam I added some docs

**Comment by pappasam** [comment]
Looks great, once the tests pass I'll be happy to merge! You can run `make test` locally to run all relevant tests locally.

**Review by pappasam — APPROVED** [review]
lgtm, thanks for all the effort!

**Pull request merged**